cmd/notary-signer/main.go

```diff
@@ -74,17 +80,41 @@ func passphraseRetriever(keyName, alias string, createNew bool, attempts int) (p
 // mapping
 func setUpCryptoservices(configuration *viper.Viper, allowedBackends []string) (
 	signer.CryptoServiceIndex, error) {
-
-	storeConfig, err := utils.ParseStorage(configuration, allowedBackends)
-	if err != nil {
-		return nil, err
-	}
+	backend := configuration.GetString("storage.backend")
 
 	var keyStore trustmanager.KeyStore
-	if storeConfig.Backend == utils.MemoryBackend {
+	switch backend {
+	case notary.MemoryBackend:
 		keyStore = trustmanager.NewKeyMemoryStore(
 			passphrase.ConstantRetriever("memory-db-ignore"))
-	} else {
+	case notary.RethinkDBBackend:
+		var sess *gorethink.Session
+		storeConfig, err := utils.ParseRethinkDBStorage(configuration)
+		if err != nil {
+			return nil, err
+		}
+		sess, err = rethinkdb.Connection(storeConfig.CA, storeConfig.Source)
+		if err == nil {
+			defaultAlias := configuration.GetString("storage.default_alias")
+			if defaultAlias == "" {
+				// backwards compatibility - support this environment variable
+				defaultAlias = configuration.GetString(defaultAliasEnv)
+			}
+
+			if defaultAlias == "" {
+				return nil, fmt.Errorf("must provide a default alias for the key DB")
+			}
+			logrus.Debug("Default Alias: ", defaultAlias)
+			s := keydbstore.NewRethinkDBKeyStore(passphraseRetriever, defaultAlias, sess)
+			keyStore = s
+			health.RegisterPeriodicFunc("DB operational", s.CheckHealth, time.Minute)
+		}
+
+	case notary.MySQLBackend, notary.SQLiteBackend:
+		storeConfig, err := utils.ParseSQLStorage(configuration)
+		if err != nil {
+			return nil, err
+		}
 		defaultAlias := configuration.GetString("storage.default_alias")
 		if defaultAlias == "" {
```

Comment:
Ah right, forgot the positive test issues.  Maybe we can just have the negative test and check the error text for default alias info?